### DISCUSSION TITLE
Is it possible to remove the canvas dependency?

### DISCUSSION THREAD

**Discussion by axetroy** [fabricjs/fabric.js#8454] [Q&A] [answered]
```json
    "fabric": {
      "version": "5.2.1",
      "resolved": "https://registry.npmmirror.com/fabric/-/fabric-5.2.1.tgz",
      "integrity": "sha512-Irltx4i+aLccWgdQj2Uvrwh/XulDAqqYMZ1bI13fAtmlxl4ggobo0t7VVYy3Ob4YEB0sCeJZKE8ExZgGo/amkw==",
      "requires": {
        "canvas": "^2.8.0",
        "jsdom": "^19.0.0"
      }
    },
```

It uses node-gpy, which on my installation keeps failing or taking a lot of time for build from source.

I only need it to work on the web platform, but it does have to install the node addon

**Comment by axetroy** [comment]
Is it possible to split fabric into several packages if the node environment requires it.

eg:

fabric-core
fabric-dom
fabric-node

```js
// use in node.js
import { fabric } from 'fabric'
import { Canvas } from 'fabric-node'

fabric.setCanvasAdapter(Canvas)
```

```js
// use in browser
import { fabric } from 'fabric'
import { Canvas } from 'fabric-dom'

fabric.setCanvasAdapter(Canvas)
```

This, make fabric become to a cross-platform render library.

Users can implement canvas by themselves to adapt to other platforms

**Comment by ShaMan123** [comment, marked as answer]
You are barging into an open door 
#8208

**Comment by ShaMan123** [comment]
look at #8444 as well

**Comment by axetroy** [comment]
Thanks ShaMan123

**Reply by ShaMan123** [reply]
Happy to help!
BTW I see you speak Chinese, if you are interested in contributing to vertical text join here #8466

**Reply by axetroy** [reply]
> Happy to help! BTW I see you speak Chinese, if you are interested in contributing to vertical text join here #8466

I am sorry. I did not good at for this. 😞
